hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/FilterParams.java

```diff
@@ -28,34 +33,48 @@
     private final String[] filterByTagNames;
     private final Long filterByDistributionId;
 
+    private FilterParams(@NotNull Long filterByDistributionId) {
+        this(Collections.emptyList(), null, null, filterByDistributionId, null);
+    }
+
     /**
-     * Constructor.
+     * Constructor for creating filter-objects.
      *
-     * @param filterByInstalledOrAssignedDistributionSetId
-     *            if set, a filter is added for the given
-     *            {@link DistributionSet#getId()}
      * @param filterByStatus
      *            if set, a filter is added for target states included by the
      *            collection
      * @param overdueState
      *            if set, a filter is added for overdued devices
      * @param filterBySearchText
-     *            if set, a filter is added for the given search text
+     *            to find targets having the text anywhere in name or
+     *            description.
+     * @param filterByInstalledOrAssignedDistributionSetId
+     *            filter by installed or assigned
+     *            {@link DistributionSet#getId()}
      * @param selectTargetWithNoTag
-     *            if set, tag-filtering is enabled
+     *            flag to select targets with no tag assigned, if set,
+     *            tag-filtering is enabled
      * @param filterByTagNames
      *            if tag-filtering is enabled, a filter is added for the given
      *            tag-names
      */
     public FilterParams(final Collection<TargetUpdateStatus> filterByStatus, final Boolean overdueState,
             final String filterBySearchText, final Long filterByInstalledOrAssignedDistributionSetId,
             final Boolean selectTargetWithNoTag, final String... filterByTagNames) {
-        this.filterByStatus = filterByStatus;
+        this.filterByStatus = (filterByStatus == null) ? Collections.emptyList() : filterByStatus;
         this.overdueState = overdueState;
         this.filterBySearchText = filterBySearchText;
         this.filterByDistributionId = filterByInstalledOrAssignedDistributionSetId;
-        this.selectTargetWithNoTag = selectTargetWithNoTag;
-        this.filterByTagNames = filterByTagNames;
+        this.selectTargetWithNoTag = (selectTargetWithNoTag == null) ? false : selectTargetWithNoTag;
```

Comment:
double check assignments, maybe the null is checked somewhere else and means something. Also, check semantics, why is `false` the default here?

Reply:
false is appropriate as the original logic was something like `return ...  selectTargetWithNoTag != null && selectTargetWithNoTag.equals( Boolean.TRUE) ... `